test/functional/testlib.bash

```diff
@@ -2659,6 +2694,429 @@ create_bundle() { # swupd_function
 
 }
 
+add_content_to_bundle() {
+
+	actb_usage() {
+		cat <<-EOM
+		Usage:
+		    add_content_to_bundle <manifest> <content_directory>
+		EOM
+	}
+
+	# user provided
+	local manifest=$1
+	local content_dir=$2
+	validate_item "$manifest"
+	validate_path "$content_dir"
+
+	# to be used by recursive action only
+	local recursive_path=$3
+
+	# local variables
+	local bundle_name="${manifest#*Manifest.}"
+	local version_dir
+	local files_dir
+	local relative_path
+	local filehash
+	local filename
+	local content
+
+	# this function is recursive, and uses the "recursive_path" argument
+	# as feedback of where in the process the function is, if the value
+	# of recursive_path is empty it means it is the very first iteration
+	if [ -z "$recursive_path" ]; then
+		recursive_path="$content_dir"
+	fi
+
+	# shortcut to useful directories
+	version_dir=$(dirname "$manifest")
+	files_dir="$version_dir"/files
+
+	# get an array with all the content we want to add to the manifest
+	# from the current recursive path
+	mapfile -t content < <(ls "$recursive_path")
```

Comment:
Can you use find here? So we don't need to make this function recursive

Reply:
it may work, it never occurred to me I could use find for that, it is worth trying